lib/Service/Appointments/MailService.php

```diff
@@ -0,0 +1,178 @@
+	/** @var IMailer */
+	private $mailer;
+	/** @var IL10N */
+	private $l10n;
+	/** @var Defaults */
+	private $defaults;
+	/** @var LoggerInterface */
+	private $logger;
+	/** @var URLGenerator */
+	private $urlGenerator;
+	/** @var IDateTimeFormatter */
+	private $dateFormatter;
+	/** @var IFactory */
+	private $lFactory;
+
+	public function __construct(IMailer            $mailer,
+								IUserManager       $userManager,
+								IL10N              $l10n,
+								Defaults           $defaults,
+								LoggerInterface            $logger,
+								URLGenerator       $urlGenerator,
+								IDateTimeFormatter $dateFormatter,
+								IFactory           $lFactory) {
+
+		$this->userManager = $userManager;
+		$this->mailer = $mailer;
+		$this->l10n = $l10n;
+		$this->defaults = $defaults;
+		$this->logger = $logger;
+		$this->urlGenerator = $urlGenerator;
+		$this->dateFormatter = $dateFormatter;
+		$this->lFactory = $lFactory;
+	}
+
+	/**
+	 * @param Booking $booking
+	 * @param AppointmentConfig $config
+	 * @throws ServiceException
+	 */
+	public function sendConfirmationEmail(Booking $booking, AppointmentConfig $config): void {
+		$user = $this->userManager->get($config->getUserId());
+
+		if($user === null) {
+			throw new ServiceException('Could not find organizer');
+		}
+
+		$fromEmail = $user->getEMailAddress();
+		$fromName = $user->getDisplayName();
+
+
+		$sys = $this->getSysEmail();
+		$message = $this->mailer->createMessage()
+			->setFrom([$sys => $fromName])
+			->setTo([$booking->getEmail() => $booking->getDisplayName()])
+			->setReplyTo([$fromEmail => $fromName]);
+
+
+		$template = $this->mailer->createEMailTemplate('calendar.confirmAppointment');
+		$template->addHeader();
+
+		//Subject
+		$subject = $this->l10n->t('Your Appointment "%s" needs confirmation', [$config->getName()]);
+		$template->setSubject($subject);
+
+		// Heading
+		$summary = $this->l10n->t("Dear %s, please confirm your booking", [$booking->getDisplayName()]);
+		$template->addHeading($summary);
+
+		// Create Booking overview
+		$this->addBulletList($template, $this->l10n, $booking, $config->getLocation());
+
+		$bookingUrl = $this->urlGenerator->linkToRouteAbsolute('calendar.booking.confirmBooking', ['token' => $booking->getToken()]);
+		$template->addBodyButton($this->l10n->t('Confirm'), $bookingUrl);
+
+		$bodyText = $this->l10n->t('This confirmation link expires in %s hours.', [(BookingService::EXPIRY /3600)] );
+		$template->addBodyText($bodyText);
+
+		$bodyText = $this->l10n->t("If you wish to cancel the appointment after all, please contact your organizer.");
+		$template->addBodyText($bodyText);
+
+		$template->addFooter();
+
+		$message->useTemplate($template);
+
+
+		try {
+			$failed = $this->mailer->send($message);
+			if (count($failed) > 0) {
+				$this->logger->warning('Mail delivery failed for some recipients.');
+				foreach($failed as $fail) {
+					$this->logger->debug('Failed to deliver email to ' . $fail);
+				}
+			}
```

Comment:
I think you want to throw your own exception here because if you reach this line it means no email was sent out. We only have one email recipient.